src/generateSummary.ts

```diff
@@ -0,0 +1,54 @@
+import * as fs from 'fs';
+
+import { getSafeL2SingletonDeployment, getSafeSingletonDeployment } from './safes';
+import { getProxyFactoryDeployment } from './factories';
+import { getDefaultCallbackHandlerDeployment, getCompatibilityFallbackHandlerDeployment, getFallbackHandlerDeployment } from './handler';
+import { getMultiSendDeployment, getMultiSendCallOnlyDeployment, getCreateCallDeployment, getSignMessageLibDeployment } from './libs';
+import ALL_CHAINS from '../chains.json';
+
+const title = '# Safe Deployments\n';
+const description = 'This table contains a list of deployed Safe contracts. Chain information was taken from [Ethereum Lists: Chains](https://github.com/ethereum-lists/chains).\n';
+const versions = ['1.3.0', '1.2.0', '1.1.1', '1.0.0'];
+
+const headerRow = `| **Chain**                   | ${versions.map((version) => `**${version}**`.padEnd(27, ' ')).join('|')} |`;
+const seperatorRow = `| ${'-'.repeat(28)} | ${versions.map(() => ` ${'-'.repeat(26)} |`).join('')} `;
+
+const deployments = [getProxyFactoryDeployment, getSafeSingletonDeployment, getSafeL2SingletonDeployment,
+     getDefaultCallbackHandlerDeployment, getCompatibilityFallbackHandlerDeployment, getFallbackHandlerDeployment,
+     getMultiSendDeployment, getMultiSendCallOnlyDeployment, getCreateCallDeployment, getSignMessageLibDeployment];
+
+const chainRows = ALL_CHAINS
```

Comment:
The current process is (if I understand correctly):
- Predefine all chains
- Predefine all versions
- Predefine all deployments
- Iterate nested over each to collect data
- Render md file.


This is quite a lot of predefined/hardcoded logic, which imo is not necessary.

It should be possible to do the following:
- load subfolders of `assets` -> versions
- load files of subfolders -> deployments
- Collect information into map of chain id to deployment
- Load chain information async
- Render md file

This way it would automatically work when new information is added (like a new version or new chain)

Reply:
Done: https://github.com/safe-global/safe-deployments/pull/228/commits/e50a93cdade1adf5e7dfa34e33dd164b25eefb48